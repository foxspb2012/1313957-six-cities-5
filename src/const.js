export const MAXIMUM_OFFERS = 3;

export const OfferType = {
  APARTMENT: `apartment`,
  HOTEL: `hotel`,
  HOUSE: `house`,
  ROOM: `room`,
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`,
  NOT_FOUND: `/not-found`,
};

export const City = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`
};

export const MAX_RATING = 5;

export const RATING_COEFFICIENT = 20;

export const SortType = {
  POPULAR: `popular`,
  TO_HIGH_PRICE: `price: low to high`,
  TO_LOW_PRICE: `price: high to low`,
  TOP_RATED: `top rated first`,
};

export const DEFAULT_CITY = City.PARIS;

export const DEFAULT_SORT_TYPE = SortType.POPULAR;

export const NameSpace = {
  APP: `APP`,
  USER: `USER`,
  OFFERS: `OFFERS`,
  REVIEWS: `REVIEWS`,
};

export const EMPTY_CLASS = {
  MAIN: `page__main--index-empty`,
  CITIES: `cities__places-container--empty`,
};

export const ERROR_TIMEOUT = 4000;

export const ServerResponseStatus = {
  OK: 200,
  UNAUTHORIZED: 401,
};

export const ServerURL = {
  LOGIN: `/login`,
  OFFERS: `/hotels`,
  FAVORITES: `/favorite`,
  REVIEWS: `/comments`,
};

export const AuthorizationStatus = {
  AUTHORIZED: `AUTHORIZED`,
  UNAUTHORIZED: `UNAUTHORIZED`,
};
