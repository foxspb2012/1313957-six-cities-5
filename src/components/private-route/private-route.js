import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Route, Redirect} from 'react-router-dom';
import {AuthorizationStatus} from '../../const';

const PrivateRoute = (props) => {
  const {render, path, exact, authorizationStatus} = props;
  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render(routeProps)
            : <Redirect to="/login"/>
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  render: PropTypes.func,
  path: PropTypes.string,
  exact: PropTypes.bool,
  authorizationStatus: PropTypes.string
};

const mapStateToProps = (state) => ({
  authorizationStatus: state.USER.authorizationStatus
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);