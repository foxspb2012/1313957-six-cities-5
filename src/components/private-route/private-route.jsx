import React from 'react';
import {connect} from 'react-redux';
import * as Type from '../../prop-types';
import {Route, Redirect} from 'react-router-dom';
import {AppRoute, AuthorizationStatus} from '../../const';
import {getAuthorizationStatus} from '../../store/user/selectors';

const PrivateRoute = ({authorizationStatus, path, exact, render}) => {
  if (authorizationStatus === AuthorizationStatus.AUTHORIZED && path === AppRoute.LOGIN) {
    return <Redirect to={AppRoute.MAIN} />;
  }
  if (authorizationStatus !== AuthorizationStatus.AUTHORIZED && path !== AppRoute.LOGIN) {
    return <Redirect to={AppRoute.LOGIN} />;
  }
  return (
    <Route path={path} exact={exact} > {render()} </Route>
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: Type.AUTHORIZATION_STATUS.isRequired,
  path: Type.PATH.isRequired,
  exact: Type.EXACT.isRequired,
  render: Type.FUNCTION.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
