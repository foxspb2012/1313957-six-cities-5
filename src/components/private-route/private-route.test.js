import React from 'react';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route';
import {BrowserRouter} from 'react-router-dom';
import {AppRoute, AuthorizationStatus} from '../../const';

const testing = () => {};

describe(`PrivateRoute render`, () => {
  it(`PrivateRoute with authorization`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <PrivateRoute
              path={AppRoute.FAVORITES}
              exact={true}
              authorizationStatus={AuthorizationStatus.AUTHORIZED}
              onFavoriteButtonClick={testing}
              render={testing}>
            </PrivateRoute>
          </BrowserRouter>
      )

    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`PrivateRoute without authorization`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <PrivateRoute
              path={AppRoute.LOGIN}
              exact={true}
              authorizationStatus={AuthorizationStatus.UNAUTHORIZED}
              onFavoriteButtonClick={testing}
              render={testing}>
            </PrivateRoute>
          </BrowserRouter>
      )

    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
