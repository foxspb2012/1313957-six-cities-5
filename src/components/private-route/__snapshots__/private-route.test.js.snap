// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute render PrivateRoute with authorization 1`] = `null`;

exports[`PrivateRoute render PrivateRoute without authorization 1`] = `null`;
